File: Problems/No.100_Same_Tree/same_tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    
    bool isSameTree(TreeNode* p, TreeNode* q) {
        
        // start compute, using inorder traversal method
        if (p!=NULL and q!=NULL) return isSameTree(p->left, q->left) and (p->val==q->val) and isSameTree(p->right, q->right);
        else if (p==NULL and q==NULL) return true;
        else return false;
        
    }
};

File: Problems/No.100_Same_Tree/same_tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def isSameTree(self, p, q):
        """
        :type p: TreeNode
        :type q: TreeNode
        :rtype: bool
        """
        
        # start compute, using inorder traversal method
        if (p!=None and q!=None):
            return self.isSameTree(p.left, q.left) and (p.val==q.val) and self.isSameTree(p.right, q.right)
        elif (p==None and q==None):
            return True
        else:
            return False
        
